In [ ]:
https://misbar.com/v1/factcheck/listArticles?language=Arabic&page=1&limit=15&scat=&result=misleading مضلل
https://misbar.com/v1/factcheck/listArticles?language=Arabic&page=2&limit=15&scat=&result=commotion اثارة 
https://misbar.com/v1/factcheck/listArticles?language=Arabic&page=1&limit=15&scat=&result=false زائف
https://misbar.com/v1/factcheck/listArticles?language=Arabic&page=1&limit=15&scat=&result=selective انتقائي
https://misbar.com/v1/factcheck/listArticles?language=Arabic&page=1&limit=15&scat=&result=unclear مشكوك فيه
https://misbar.com/v1/factcheck/listArticles?language=Arabic&page=1&limit=15&scat=&result=true صحيح
https://misbar.com/v1/factcheck/listArticles?language=Arabic&page=0&limit=15&scat=&result=myth  خرافة
https://misbar.com/v1/factcheck/listArticles?language=Arabic&page=1&limit=15&scat=&result=satire ساخر

In [ ]:
import urllib.request as urllib2
import json
import csv
from urllib.parse import unquote,quote


MainUrl="https://misbar.com/v1/factcheck/listArticles?language=Arabic"
pages = {"misleading","commotion","false","selective","unclear","true","myth","satire"}

for x in pages:
  count=0
  csvfile = open('drive/My Drive/Arabic Fake news/Misbar/csvfiles/'+x+'.csv', 'w+', newline='')
  writer = csv.writer(csvfile)
  for i in range(100):
    URL = MainUrl+"&page="+str(i)+"&limit=15&scat=&result="+x
    response = urllib2.urlopen(URL)
    data = json.loads(response.read())
    if len(data) !=0 :
      for item in data['data']:
        if item['url_path']:
          href="https://misbar.com/"+quote(item['url_path'])
          writer.writerow([href])
          count=count+1
  print(str(x)+".csv count = "+str(count))
  csvfile.close()

print("Finish")


commotion.csv count = 61
myth.csv count = 18
selective.csv count = 29
unclear.csv count = 27
satire.csv count = 45
true.csv count = 66
misleading.csv count = 1440
false.csv count = 1040
Finish


1. commotion.csv count = 61
2. myth.csv count = 18
3. selective.csv count = 29
4. unclear.csv count = 27
5. satire.csv count = 45
6. true.csv count = 66
7. misleading.csv count = 1440
8. false.csv count = 1040
Finish

In [ ]:
pip install -q newspaper3k

In [ ]:
from newspaper import Article
import csv
import json
import os

pages = {
  "misleading": 'drive/My Drive/Arabic Fake news/Misbar/csvfiles/misleading.csv',
  "commotion": 'drive/My Drive/Arabic Fake news/Misbar/csvfiles/commotion.csv',
  "false":'drive/My Drive/Arabic Fake news/Misbar/csvfiles/false.csv',
  "selective": 'drive/My Drive/Arabic Fake news/Misbar/csvfiles/selective.csv',
  "unclear": 'drive/My Drive/Arabic Fake news/Misbar/csvfiles/unclear.csv',
  "true":'drive/My Drive/Arabic Fake news/Misbar/csvfiles/true.csv',
  "myth": 'drive/My Drive/Arabic Fake news/Misbar/csvfiles/myth.csv',
  "satire":'drive/My Drive/Arabic Fake news/Misbar/csvfiles/satire.csv'
}
for x in pages:
  basePath='drive/My Drive/Arabic Fake news/Misbar/RawData/'+str(x)+'/'
  if not os.path.exists(basePath):
    os.makedirs(basePath)
  i=0  
  file = open(pages[x])
  csv_file = csv.reader(file)
  for link in csv_file:
    i=i+1
    link='/'.join(link[0].split("/")[-4:])
    link='https://misbar.com/v1/factcheck/getBySlug/'+str(link)
    response = urllib2.urlopen(link)
    data = json.loads(response.read())
    c_data=data['data']
    with open(basePath+str(x)+str(i)+'.json','w',encoding='utf8') as outfile:
      json.dump(c_data, outfile, indent=4,sort_keys=True, default=str,ensure_ascii=False)
  print("Finish Collecting " +str(x)+" "+str(i) +" news ") 

Finish Collecting misleading 1440 news 
Finish Collecting commotion 61 news 
Finish Collecting false 1040 news 
Finish Collecting selective 29 news 
Finish Collecting unclear 27 news 
Finish Collecting true 66 news 
Finish Collecting myth 18 news 
Finish Collecting satire 45 news 


Finish Collecting misleading 1440 news 
Finish Collecting commotion 61 news 
Finish Collecting false 1040 news 
Finish Collecting selective 29 news 
Finish Collecting unclear 27 news 
Finish Collecting true 66 news 
Finish Collecting myth 18 news 
Finish Collecting satire 45 news 

In [ ]:
pip install -q XlsxWriter

     |████████████████████████████████| 153kB 12.9MB/s 


In [ ]:
from io import StringIO
from html.parser import HTMLParser

class MLStripper(HTMLParser):
    def __init__(self):
        super().__init__()
        self.reset()
        self.strict = False
        self.convert_charrefs= True
        self.text = StringIO()
    def handle_data(self, d):
        self.text.write(d)
    def get_data(self):
        return self.text.getvalue()

def strip_tags(html):
    s = MLStripper()
    s.feed(html)
    return s.get_data()

In [ ]:
import urllib.request as urllib2
from urllib.parse import unquote,quote
import xlsxwriter
import json
import os
import csv

pages = {
  "misleading": 'drive/My Drive/Arabic Fake news/Misbar/csvfiles/misleading.csv',
  "commotion": 'drive/My Drive/Arabic Fake news/Misbar/csvfiles/commotion.csv',
  "false":'drive/My Drive/Arabic Fake news/Misbar/csvfiles/false.csv',
  "selective": 'drive/My Drive/Arabic Fake news/Misbar/csvfiles/selective.csv',
  "unclear": 'drive/My Drive/Arabic Fake news/Misbar/csvfiles/unclear.csv',
  "true":'drive/My Drive/Arabic Fake news/Misbar/csvfiles/true.csv',
  "myth": 'drive/My Drive/Arabic Fake news/Misbar/csvfiles/myth.csv',
  "satire":'drive/My Drive/Arabic Fake news/Misbar/csvfiles/satire.csv'}

for x in pages:
  basePath='drive/My Drive/Arabic Fake news/Misbar/XlsxData/OriginalData/'+str(x)+'.xlsx'
  workbook = xlsxwriter.Workbook(basePath)
  worksheet = workbook.add_worksheet()

  file = open(pages[x])
  csv_file = csv.reader(file)
  i=0 

  for link in csv_file:
    link='/'.join(link[0].split("/")[-4:])
    link='https://misbar.com/v1/factcheck/getBySlug/'+str(link)
    response = urllib2.urlopen(link)
    data = json.loads(response.read())
    c_data=data['data']

    claim=strip_tags(c_data["claim"])
    title=c_data["title"]
    category=c_data["category"]
    #caption=c_data["caption"]
    url_path="https://misbar.com/"+quote(c_data['url_path'])

    recored=[title,claim,category,url_path]

    worksheet.write_row(i, 0, recored)
    i=i+1
  workbook.close()
  print("Finish Collecting " +str(x)+" "+str(i) +" news ") 



Finish Collecting misleading 1440 news 
Finish Collecting commotion 61 news 
Finish Collecting false 1040 news 
Finish Collecting selective 29 news 
Finish Collecting unclear 27 news 
Finish Collecting true 66 news 
Finish Collecting myth 18 news 
Finish Collecting satire 45 news 
